File: Arrays & Hashing/2485. Find the Pivot Integer/2485.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int prefix = 0;
        vector<int> prefixSum;
        for (int i=0; i<n+1; ++i) {
            prefix += i;
            prefixSum.push_back(prefix);
        } prefix = 0;
        
        for (int i=n; i>-1; --i) {
            prefix += i;
            if (prefix==prefixSum[i]) {
                return i;
            }
        } return -1;
    }
};

File: Arrays & Hashing/2485. Find the Pivot Integer/2485.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        prefix = 0
        prefix_sum = [i for i in range(n + 1)]
        for i in range(n + 1):
            prefix_sum[i] += prefix
            prefix += i
        
        prefix = 0
        for i in range(n, -1, -1):
            prefix += i
            if prefix_sum[i] == prefix:
                return i
        return -1

        